File: core/src/androidTest/mock/com/mobilesolutionworks/android/controller/test/sdk/RetainChildFragmentActivity.java
package com.mobilesolutionworks.android.controller.test.sdk;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.mobilesolutionworks.android.controller.test.R;

/**
 * Created by yunarta on 9/3/17.
 */

public class RetainChildFragmentActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_host_activity);
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getSupportFragmentManager().beginTransaction()
                        .addToBackStack("back-stack")
                        .replace(R.id.fragment_container, new EmptyFragment(), "stack")
                        .commit();
            }
        });

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_container, new RootFragment(), "root")
                    .commitNow();
        }
    }
}

File: test-app/src/main/kotlin/com/mobilesolutionworks/android/controller/test/sdk/RetainChildFragmentActivity.kt
package com.mobilesolutionworks.android.controller.test.sdk

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import com.mobilesolutionworks.android.controller.test.R

/**
 * Created by yunarta on 9/3/17.
 */

public class RetainChildFragmentActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_host_activity)
        findViewById<Button>(R.id.button).setOnClickListener {
            supportFragmentManager.beginTransaction()
                  .addToBackStack("back-stack")
                  .replace(R.id.fragment_container, EmptyFragment(), "stack")
                  .commit()
        }

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                  .replace(R.id.fragment_container, RootFragment(), "root")
                  .commitNow()
        }
    }
}
